### PR TITLE
obs-ffmpeg: Receive packets while frame queue is full

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

Previously we always assumed we could push more frames before an encoded packet would be available. Actually we can get EAGAIN when the frame queue is full and before any encoded packets have been completed.

This changes attempts to read packets more often as the worst that can happen is we get EAGAIN again (i guess).

### Motivation and Context

Fixes a hang in ffmpeg vaapi encoder on intel.

### How Has This Been Tested?

The resulting recording looks correct after enabling bframes on intel's av1 drivers.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c [hunk 1]
```diff
@@ -543,7 +543,7 @@ static bool vaapi_encode(void *data, struct encoder_frame *frame,
 	}
 
 	ret = avcodec_send_frame(enc->context, hwframe);
-	if (ret == 0)
+	if (ret == 0 || AVERROR(EAGAIN))
```
Comment (notr1ch) [inline plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c:546]: Shouldn't this be `ret == AVERROR(EAGAIN)`?

Comment (kkartaltepe) [reply plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c]: That is what I thought I typed, fixed and retested. Thanks.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Commit message nits:
* Previously -> Previously,
* Actually -> Actually,
* This changes -?> This change

The first two are just grammar nits. The last one I wasn't sure if it meant to say "This change" or not, hence the question mark.

As this is contained to VAAPI, I'm otherwise fine with the PR.

**Comment by kkartaltepe** [comment]
Fixed up the commit message.

**Review by Lain-B — APPROVED** [review]

**Pull request merged**